packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1386,27 +1394,185 @@ class SemanticsConfiguration {
   /// See also:
   ///
   /// * [addAction] to add an action.
-  final Map<SemanticsAction, VoidCallback> _actions = <SemanticsAction, VoidCallback>{};
+  final Map<SemanticsAction, _SemanticsActionHandler> _actions = <SemanticsAction, _SemanticsActionHandler>{};
 
   int _actionsAsBits = 0;
 
   /// Adds an `action` to the semantics tree.
   ///
   /// Whenever the user performs `action` the provided `handler` is called.
-  void addAction(SemanticsAction action, VoidCallback handler) {
+  void _addAction(SemanticsAction action, _SemanticsActionHandler handler) {
     assert(handler != null);
     _actions[action] = handler;
     _actionsAsBits |= action.index;
     _hasBeenAnnotated = true;
   }
 
+  /// Adds an `action` to the semantics tree, whose `handler` does not expect
+  /// any arguments.
+  ///
+  /// Whenever the user performs `action` the provided `handler` is called.
+  void _addArgumentlessAction(SemanticsAction action, VoidCallback handler) {
+    assert(handler != null);
+    _addAction(action, (dynamic args) {
+      assert(args == null);
+      handler();
+    });
+  }
+
+  /// The handler for [SemanticsAction.tap].
+  ///
+  /// This is the semantic equivalent of a user briefly tapping the screen with
+  /// the finger without moving it. For example, a button should implement this
+  /// action.
+  ///
+  /// VoiceOver users on iOS and TalkBack users on Android can trigger this
+  /// action by double-tapping the screen while an element is focused.
+  set onTap(VoidCallback handler) {
```

Comment:
It is against the style guide to provide a setter without a getter. Can we provide getters? If not, can we make these methods instead of setters?

Reply:
I tried to find this rule in our style guide, but wasn't able to locate it. Can you link me to it?

Reply:
If it's not in the style guide, it certainly should be. :-)

Reply:
I added the getters here and down below in this file where they were also missing.

Reply:
I also updated the link in Flutter's style guide to point to the correct landing page of Dart's style guide. With that, the rule Yegor mentioned is now included in the transitive closure of our style guides.